### PR TITLE
feat: wds color generation algorithm

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description

1. Implementation of the [POC6](https://www.figma.com/file/4hEEIIZngaovgRg6UgUq4j/Color-Generator-POC6%3A-If%2FElse-variations%2C-multiple-widget-application%2C-light-and-dark-mode?node-id=0-1&t=EgOsrdXPpR19wCt8-0) algorithm
2. Adding a component to test the algorithm
3. Adding a light and dark theme switch to Storybook

Fixes # (issue)
#21928

Media
https://user-images.githubusercontent.com/11555074/231161246-1641a41a-2e85-4b88-99f7-7e69b4d7de18.mov

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/Button.tsx [hunk 1]
```diff
@@ -11,12 +11,15 @@ import type { ButtonProps as SpectrumButtonProps } from "@react-types/button";
 
 export interface ButtonProps extends SpectrumButtonProps {
   className?: string;
+  isActive?: boolean;
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Button.tsx:14]: I'll delete all this as soon as we finish the color generation

https://github.com/appsmithorg/appsmith/issues/22286

**Review by ichik — APPROVED** [review]

File: app/client/packages/design-system/theming/src/utils/colorUtils/setLch.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import type Color from "colorjs.io";
+import { parse } from "./parse";
+import type { ColorTypes } from "colorjs.io/types/src/color";
+
+// Ratio to replace chroma values from Figma(0-50) to colorjs(0-0.4)
+const CHROMA_RATIO = 0.008;
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/utils/colorUtils/setLch.ts:6]: I added these variables to reduce the cognitive load when transferring values from the Figma plugin. As soon as we finish the algorithm, we will be able to delete these variables.

File: app/client/packages/design-system/theming/src/components/ThemeProvider.tsx [hunk 3]
```diff
@@ -30,7 +29,11 @@ const StyledProvider = styled.div<ThemeProviderProps>`
 `;
 
 export const ThemeProvider = (props: ThemeProviderProps) => {
-  const { children, theme } = props;
+  const { children, className, theme } = props;
```
Comment (jsartisan) [inline app/client/packages/design-system/theming/src/components/ThemeProvider.tsx:33]: btw if you are passing all the coming props to component, you can simply do 

 <StyledProvider {...props} />

File: app/client/packages/design-system/theming/src/tokens/defaultTokens.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "seedColor": "#553de9",
   "focusColor": "#2a82ea",
   "rootUnit": 4,
-  "isDarkMode": false,
+  "colorScheme": "light",
```
Comment (jsartisan) [inline app/client/packages/design-system/theming/src/tokens/defaultTokens.json:5]: what are all possible values of colorScheme we are thinking for phase 1?

File: app/client/packages/design-system/widgets/src/components/Text/Text.stories.mdx [hunk 5]
```diff
@@ -25,7 +28,7 @@ Text is a component that renders a capsized text.
 
 <ArgsTable of={Text} story="Text" />
 
-# Trucated text
+# Truncated text
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/Text/Text.stories.mdx:31]: thanks for fixing the typo :P

**Review by jsartisan — DISMISSED** [review]
LGTM, left 2-3 minor questions/comments.

File: app/client/packages/design-system/theming/src/tokens/defaultTokens.json (hunk 4 above)
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/tokens/defaultTokens.json:5]: We are only working on the dark and light schemes now.

File: app/client/packages/design-system/theming/src/components/ThemeProvider.tsx (hunk 3 above)
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/components/ThemeProvider.tsx:33]: I believe the current implementation is more verbose.

File: app/client/packages/design-system/theming/src/utils/colorUtils/setLch.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/utils/colorUtils/setLch.ts:6]: I deleted variables in some cases they confused.

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Pull request merged**